client/ServerRunner.cpp

```diff
@@ -23,19 +23,30 @@
 ServerThreadRunner::ServerThreadRunner() = default;
 ServerThreadRunner::~ServerThreadRunner() = default;
 
-void ServerThreadRunner::start(uint16_t port, bool connectToLobby, std::shared_ptr<StartInfo> startingInfo)
+uint16_t ServerThreadRunner::start(uint16_t port, bool connectToLobby, std::shared_ptr<StartInfo> startingInfo)
 {
-	server = std::make_unique<CVCMIServer>(port, connectToLobby, true);
+	server = std::make_unique<CVCMIServer>(port, true);
 
 	if (startingInfo)
 	{
 		server->si = startingInfo; //Else use default
 	}
 
-	threadRunLocalServer = boost::thread([this]{
+	uint16_t srvport = port;
+
+	threadRunLocalServer = boost::thread([this, connectToLobby, &srvport]{
 		setThreadName("runServer");
+		srvport = server->prepare(connectToLobby);
 		server->run();
 	});
+
+	while(srvport == 0) {
+		logNetwork->trace("Waiting for server port...");
+		boost::this_thread::sleep(boost::posix_time::milliseconds(100));
+		logNetwork->debug("Server port: %d", srvport);
+	}
```

Comment:
Can we do something better than sleep-based thread synchronization? std::future for example? Or anything else that is not sleep?

Reply:
Sure, I will look into that.